scripts/setup_env_linux_debian.sh

```diff
@@ -0,0 +1,38 @@
+#!/bin/bash
+# setup dev env on ubuntu
+
+# check if sudo permission is not available
+if [ "$EUID" -ne 0 ]; then
+    echo "sudo permission is not available"
+    exit
+fi
+
+# install system-wide dependencies 
+apt-get install python3.6 python-pip
+pip install virtualenv
+
+virtualenv -p $(which python3.6) venv
```

Comment:
I think you should just create the virtual environment without specifying the python version virtaulenv will pick the latests or you can ask user to specif path to desired python version which should be 3.6. Because this construct leads to problems on ubuntu

Reply:
`virtualenv` doesn't pick the latest version, it picks the default one. And the application is based on `python3.6`.
What we can do is ask for the desired path for the `python3.6` binary to the user.

Reply:
@Oishika-Pradhan agreed on that. There is a problem tbh - apt-get install python3.6 might not work on some architectures (might even break some depending on how messed up their original install is). This is  why I always recommend people to use a python version manager, like pyenv. Let me paste how this part should look like (copying from my own personal project) 


```
Install pyenv using following scripts: https://github.com/pyenv/pyenv-installer. The installation requires you to add some things to your .bashrc or .zshrc file. In my case, it is:
$ curl -L https://github.com/pyenv/pyenv-installer/raw/master/bin/pyenv-installer | bash
...
export PATH="/home/user/.pyenv/bin:$PATH"
eval "$(pyenv init -)"
eval "$(pyenv virtualenv-init -)"
...
and source the rc file using source ~/.bashrc

on macos:
...
brew install pyenv
brew install pyenv-virtualenv
...
and add the following to your ~/.bash_profile (and after that do source ~/.bash_profile):

...
eval "$(pyenv init -)"
eval "$(pyenv virtualenv-init -)"
...

Now install the right python version using 
...
$ pyenv install 3.6.4 
...
Hint: If it shows something like: WARNING: The Python sqlite3 extension was not compiled. Missing the SQLite3 lib?, you can install it following something like: https://gist.github.com/antivanov/01ed4eac2d7486a170be598b5a0a4ac7. See that we are not using SQLite for local testing anymore, so you can actually ignore this.

create a virtual environment for the project: 
...
pyenv virtualenv 3.6.4 yourvenvname-p3
...
activate the virtual environment for the repository (this will create a .python-version file, which is ignored via .gitignore):
...
pyenv local yourvenvname-p3
...
- this would automatically activate the env as soon as you CD into the dir

install dependencies: 
...
pip install -r requirements-dev.txt
...
```